src/pcre2test_inc.h

```diff
@@ -564,18 +564,23 @@ return 0;
 static void
 config_str(uint32_t what, char *where)
 {
-int r1, r2;
-PCRE2_UCHAR buf[VERSION_SIZE];
+int r2;
+PCRE2_UCHAR *buf;
+int needed_len;
 
-r1 = pcre2_config(what, NULL);
-r2 = pcre2_config(what, buf);
-if (r1 < 0 || r1 != r2 || r1 >= VERSION_SIZE)
+needed_len = pcre2_config(what, NULL);
+if (needed_len <= 0)
   {
   cfprintf(clr_test_error, stderr, "pcre2test: Error in pcre2_config(%d)\n", what);
   exit(1);
   }
 
-while (r1-- > 0) where[r1] = (char)buf[r1];
+buf = malloc(needed_len * sizeof(PCRE2_UCHAR));
+r2 = pcre2_config(what, buf);
+PCRE2_ASSERT(r2 == needed_len);
+
+while (r2-- > 0) where[r2] = (char)buf[r2];
+free(buf);
```

Comment:
The caller is still using a VERSION_SIZE buffer. I think you need to malloc a char buffer and return it for the caller to free. You can then remove the VERSION_SIZE buffer.

Otherwise, also I like this PR.

Reply:
Thank you for addressing my concerns. The code here appears to be correct & working. I like it.